ibis/backends/flink/tests/conftest.py

```diff
@@ -19,11 +19,44 @@ def connect(*, tmpdir, worker_id, **kw: Any):
         comply with the assumption that the tests under ibis/ibis/backends/tests/
         are for batch (storage or processing) backends.
         """
-        from pyflink.table import EnvironmentSettings, TableEnvironment
-
-        env_settings = EnvironmentSettings.in_batch_mode()
-        table_env = TableEnvironment.create(env_settings)
-        return ibis.flink.connect(table_env, **kw)
+        import os
+
+        from pyflink.java_gateway import get_gateway
+        from pyflink.table import (
+            EnvironmentSettings,
+            StreamTableEnvironment,
+            TableEnvironment,
+        )
+        from pyflink.table.table_environment import StreamExecutionEnvironment
+
+        if not bool(os.environ.get("FLINK_REMOTE_CLUSTER", False)):
```

Comment:
Can we remove this check now?

Reply:
I feel this option is still useful that if you don't want to setup a flink docker locally, you can still run the unit tests. Otherwise we have to require developers to maintain a flink cluster in their box.

Reply:
I think it's okay to require developers to start the container.

We generally don't maintain multiple test deployment methods for any other backend, and I would like to avoid introducing that precedent.

Reply:
sure, I'll go that direction